charts/frontend/templates/prometheusrules.yaml

```diff
@@ -0,0 +1,18 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: frontend.rules
+  namespace: monitoring
+spec:
+  groups:
+  - name: frontend
+    rules:
+    - alert: FrontendNotDeployedSuccessfully
+      annotations:
+        description: "Frontend has not deployed successfully in namespace {{ "{{" }} $labels.namespace }}"
+      expr: |-
+        (kube_deployment_status_condition{deployment="frontend", condition="ReplicaFailure", status="true"} == on (namespace) 
+        kube_deployment_status_condition{deployment="frontend", condition="Progressing", status="false"})==1
```

Comment:
Since the alert is specific to $labels.namespace, you probably want to select on the namespace here, rather than match all namespaces and alert if any of them have a failed rollout.

Reply:
To clarify, are you saying we should pin this down to a particular namespace that we care about, rather than have the check performed over all namespaces and alerting on each of those namespaces if they're non-compliant?

Reply:
Yes, in particular because the context in which you're writing the alert (i.e. the Helm chart) is already specific to one namespace, so you wouldn't want it to know about anything in other namespaces.